#166873552 Change log file location

kafuuma opened the pull request:
#### What Does This PR Do?
- Implement logfile rotate for logs

#### Description Of Task To Be Completed
- Add log file to store log records
- Specify location of log file to be /var/log/andela_socials/ in production and /tmp/log/andela_socials/ in 
  development

#### How can this be manually tested?
- Add `LOG_DIR=/tmp/log/andela_socials/a_social.log` variable to dev.env file
- Start the server
- Make few requests to the server
- Navigate to `/tmp/log/andela_socials/`
- `cat  a_social.log` to view logs
#### What are the relevant pivotal tracker stories?
[#166873552](https://www.pivotaltracker.com/story/show/166873552)
#### Screenshot(s)

andrewinsoul commented on .circleci/config.yml:
```diff
@@ -1,100 +1,99 @@
+                  - TERM: xterm-256color
+                  - POSTGRES_USER: circleci
+                  - POSTGRES_DB: circlecidb
+                  - POSTGRES_PASSWORD: $DB_PASSWORD
```
The `LOG_DIR` env variable was not included and that is why probably it did not show up on CircleCi. Include that and push again

kafuuma committed:
chore(logfile):Implement logfile rotation

- Specify location of log file to be /var/log/andela_socials/ in production
- And /tmp/log/andela_socials/ in development

* fix(event-details): fix event detai

brianbirir commented on .circleci/config.yml:
```diff
@@ -1,100 +1,99 @@
+                  command: |
+                      . venv/bin/activate
+                      export DJANGO_SETTINGS_MODULE=a_socials.settings.dev
+                      python server/manage.py test --snapshot-update server/api/tests/ server/graphql_schemas/tests/  --keepdb
```
Please remove the `--keep-db` flag. The test database should ideally be deleted.

brianbirir commented on client/assets/components/navbar.scss:
```diff
@@ -2,10 +2,10 @@
 @import '../colors.scss';
```
The focus of this PR is to change log file location and not editing or updating the client side of the application. Please revert.

brianbirir commented on server/a_socials/settings/dev.py:
```diff
@@ -33,3 +33,10 @@
 
 ENVIRONMENT = "development"
+
+# load logg settings
```
Please provide correct spelling i.e. should be **log**

brianbirir commented:
@kafuuma this PR should focus on one story only but is currently associated with two Pivotal Tracker stories. The appropriate story is https://www.pivotaltracker.com/n/projects/2174978/stories/166873552.

Please change the PR title to reflect this i.e. **#166873552 Change log file location** and update the description. Also complete the section on **How can this be manually tested?** in the PR description section. Thanks.

brianbirir commented:
@kafuuma in regards to log file rotation, this should be implemented server side in the Linux environment and in Docker:

https://medium.com/@Quigley_Ja/rotating-docker-logs-keeping-your-overlay-folder-small-40cfa2155412
https://markuscisler.com/rotate-docker-logs/

For log rotate,  please create a separate PR.

brianbirir commented on docker/server/Dockerfile:
```diff
@@ -13,6 +13,13 @@ RUN pip wheel --wheel-dir=/wheelhouse -r requirements.txt
 RUN pip install --no-index -f /wheelhouse -r requirements.txt
 
+# Create directory for logs
+RUN mkdir /tmp/log &&  mkdir /tmp/log/andela_socials/
```
this can be done with one command:

```
mkdir -p /tmp/log/andela_socials
```

please update @kafuuma

brianbirir commented on server/a_socials/settings/logger_settings.py:
```diff
@@ -0,0 +1,53 @@
+import dotenv
+import os
+
+os.makedirs('/tmp/log/andela_socials', mode=0o777, exist_ok=True)
```
@kafuuma what's the purpose of this? I thought it's already been taken care of in `docker/server/Dockerfile` line 17

brianbirir commented:
Works as expected from the Docker perspective. Just a few changes need to be made @kafuuma

brianbirir commented:
Looks good.
